core/store/src/columns.rs

```diff
@@ -327,6 +350,72 @@ impl DBCol {
+            DBCol::IncomingReceipts => vec![DBKeyType::BlockHash, DBKeyType::ShardId],
+            DBCol::Peers => vec![DBKeyType::PeerId],
+            DBCol::EpochInfo => vec![DBKeyType::EpochId],
+            DBCol::BlockInfo => vec![DBKeyType::BlockHash],
+            DBCol::Chunks => vec![DBKeyType::ChunkHash],
+            DBCol::PartialChunks => vec![DBKeyType::ChunkHash],
+            DBCol::BlocksToCatchup => vec![DBKeyType::BlockHash],
+            DBCol::StateDlInfos => vec![DBKeyType::BlockHash],
+            DBCol::ChallengedBlocks => vec![DBKeyType::BlockHash],
+            DBCol::StateHeaders => vec![DBKeyType::ShardId, DBKeyType::BlockHash],
+            DBCol::InvalidChunks => vec![DBKeyType::ChunkHash],
+            DBCol::BlockExtra => vec![DBKeyType::BlockHash],
+            DBCol::BlockPerHeight => vec![DBKeyType::BlockHeight],
+            DBCol::StateParts => vec![DBKeyType::BlockHash, DBKeyType::ShardId, DBKeyType::PartId],
+            DBCol::EpochStart => vec![DBKeyType::EpochId],
+            DBCol::AccountAnnouncements => vec![DBKeyType::AccountId],
+            DBCol::NextBlockHashes => vec![DBKeyType::PreviousBlockHash],
+            DBCol::EpochLightClientBlocks => vec![DBKeyType::EpochId],
+            DBCol::ReceiptIdToShardId => vec![DBKeyType::ReceiptHash],
+            DBCol::PeerComponent => vec![DBKeyType::PeerId],
+            DBCol::ComponentEdges => vec![DBKeyType::Nonce],
+            DBCol::LastComponentNonce => vec![DBKeyType::Empty],
+            DBCol::Transactions => vec![DBKeyType::TransactionHash],
+            DBCol::StateChanges => vec![DBKeyType::BlockHash, DBKeyType::TrieKey],
+            DBCol::BlockRefCount => vec![DBKeyType::BlockHash],
+            DBCol::TrieChanges => vec![DBKeyType::BlockHash, DBKeyType::ShardId],
+            DBCol::BlockMerkleTree => vec![DBKeyType::BlockHash],
+            DBCol::ChunkHashesByHeight => vec![DBKeyType::BlockHeight],
+            DBCol::BlockOrdinal => vec![DBKeyType::BlockOrdinal],
+            DBCol::OutcomeIds => vec![DBKeyType::BlockHash, DBKeyType::ShardId],
+            DBCol::ProcessedBlockHeights => vec![DBKeyType::BlockHeight],
+            DBCol::Receipts => vec![DBKeyType::ReceiptHash],
+            DBCol::CachedContractCode => vec![DBKeyType::ContractCacheKey],
+            DBCol::EpochValidatorInfo => vec![DBKeyType::EpochId],
+            DBCol::HeaderHashesByHeight => vec![DBKeyType::BlockHeight],
+            DBCol::StateChangesForSplitStates => vec![DBKeyType::BlockHash, DBKeyType::ShardId],
+            #[cfg(feature = "protocol_feature_flat_state")]
+            DBCol::FlatState => vec![DBKeyType::TrieKey],
+            #[cfg(feature = "protocol_feature_flat_state")]
+            DBCol::FlatStateDeltas => vec![DBKeyType::ShardId, DBKeyType::BlockHash],
+            #[cfg(feature = "protocol_feature_flat_state")]
+            DBCol::FlatStateMisc => vec![DBKeyType::ShardId],
+            _ => vec![],
```

Comment:
I’d say it’s better to list the deprecated columns explicitly and then to remove this default case.  This will force people adding new column to add it to this function.